### ISSUE TITLE
Multi-select Autocomplete

### ISSUE THREAD

**Issue by stixistyson** [marmelab/react-admin#2138]
The [example](https://github.com/marmelab/react-admin/blob/0e86299a3ad7fa685a457d8b7067e93ee98bf8fc/packages/ra-ui-materialui/src/input/ReferenceArrayInput.js#L196) does not provide filtering as it is decribed in the example text. Though being the filterToQuery prop is required, it should allow for filtering automatically.

The filterToQuery function never fires being there is not area to input text.

```
field.input = props => (
    <ReferenceArrayInput
        source={field.name}
        reference={field.reference.name}
        label="Destination"
        key={field.name}
        filter={{isLeaf: true}}
        filterToQuery={searchText => {console.log('Filter the Destination: ' + searchText); return ({ displayname: searchText })}}
        sort={{field: 'displayname', order: 'ASC'}}
        {...props}>
        <SelectArrayInput optionText="displayname" optionValue="id"  />
    </ReferenceArrayInput>
);
```

This should bring up the items that are already selected, and other possible items, then as you type, filter the items so they can be selected and added to the multiselect.

If this is not the way to do this, or if more custom code should be inserted, please describe, or direct to a tutorial instead of a faulty example.

**Environment**

* React-admin version: 2.2.0
* Last version that did not exhibit the issue (if applicable): false
* React version: 16.4.1
* Browser:Chrome 68.0.3440.84
* Stack trace (in case of a JS error):

**Comment by stixistyson** [comment]
Ok, being it doesn't seem there is much discussion on this, I am guessing this is a frequently mentioned problem that no-one wants to touch. Can't blame you, I am seeing a lot of frustration since Material-ui removed the autocomplete array element.

So, being I liked the look of the ReferenceArrayInput->SelectArrayInput, I cobbled together a semi-functioning solution with a similar feel, though it does not function on a mobile device. Being the text input isn't directly targeted, the keyboard doesn't show. But it does function on a desktop, which is my main target at the moment. The filterToQuery functions as desired, and still takes the same props to get it to function.

[AutocompleteArrayInput.js](https://gist.github.com/stixistyson/b9a0fcb5bcacb63c955cde7882d01e97)

Being this is my first foray into React, I am sure there are many coding standards that I am violating/missing. I am sure there is a much easier way to update the TextInput as well, but after failing to get it targeted directly to receive input, the passthru is my solution. Hence duct-taped.

If someone can get this up and running, integrate it into one of the pulls, or take this as inspiration to bring this functionality back into this project, I'm sure many would appreciate it.

**Comment by eaverdeja** [comment]
This was my first pain point with react-admin. I admit I was kinda surprised when I realized the SelectArrayInput had no support for autocompletes. I've just begun my React adventures, so I don't feel confident enough to help (yet), but I will try out @stixistyson 's solution on the weekend.

**Comment by djhi** [comment]
Fixed by #2272
